### PR TITLE
2.x: cleanup and test renames

### PR DESCRIPTION
**Pull request by akarnokd**
This PR adds the missing `@Override` annotations and renames the `Observable` unit tests to conform a single name pattern.

### CODE REVIEW

File: src/main/java/io/reactivex/disposables/Disposable.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
     /**
      * Returns true if this resource has been disposed.
```
Comment (vanniktech) [inline src/main/java/io/reactivex/disposables/Disposable.java:25]: Is this line then really needed?

Comment (akarnokd) [inline src/main/java/io/reactivex/disposables/Disposable.java:25]: javadoc complains if the `@return` tag is missing or empty.

Comment (vanniktech) [inline src/main/java/io/reactivex/disposables/Disposable.java:25]: Yeah I know that but then `* Returns true if this resource has been disposed.` could be removed, could not it?

Comment (akarnokd) [inline src/main/java/io/reactivex/disposables/Disposable.java:25]: Yes.

**Pull request merged**